src/assets/dpos.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import {
+	getLisk32AddressFromAddress,
+	getLisk32AddressFromPublicKey,
+} from '@liskhq/lisk-cryptography';
+import {
+	MODULE_NAME_DPOS,
+	INVALID_BLS_KEY,
+	DUMMY_PROOF_OF_POSSESSION,
+	INVALID_ED25519_KEY,
+	DPOS_INIT_ROUNDS,
+	// ROUND_LENGTH,
+} from '../constants';
+
+import {
+	AccountEntry,
+	BlockEntry,
+	ModuleResponse,
+	ValidatorEntry,
+	VoterEntry,
+	GenesisDataEntry,
+} from '../types';
+
+export const getValidatorKeys = async (blocks: BlockEntry[]) => {
+	const keys = [];
+	for (const block of blocks) {
+		const lskAddress: any = getLisk32AddressFromPublicKey(block.header.generatorPublicKey);
+		keys[lskAddress] = block.header.generatorPublicKey.toString('hex');
+		for (const trx of block.payload) {
+			const trxAddress: any = getLisk32AddressFromPublicKey(trx.senderPublicKey);
+			keys[trxAddress] = trx.senderPublicKey.toString('hex');
+		}
+	}
+	return keys;
+};
```

Comment:
As per the [LIP](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0063.md#assets-entry-for-dpos-module),  `keys` must be a **dictionary** (i.e. an **object** in JS of the type `Record<string, string>`).

Could we please ensure that we explicitly define the data types for the internal properties (required) and the function return types (optional, but required wherever we expect non-standard responses i.e there's a possibility of multiple response types) wherever necessary?

```suggestion
export const getValidatorKeys = async (blocks: BlockEntry[]): Promise<Record<string, string>> => {
	const keys: Record<string, string> = {};
	for (const block of blocks) {
		const lskAddress: string = getLisk32AddressFromPublicKey(block.header.generatorPublicKey);
		keys[lskAddress] = block.header.generatorPublicKey.toString('hex');
		for (const trx of block.payload) {
			const trxAddress: string = getLisk32AddressFromPublicKey(trx.senderPublicKey);
			keys[trxAddress] = trx.senderPublicKey.toString('hex');
		}
	}
	return keys;
};
```